### PR TITLE
Feature/database migrations

### PR DESCRIPTION
**Pull request by maksymbilokur**
migration tool: golang/migrate
DDL queries in up/down migrations for table filters in database.
DCL commands in up/down migrations to provide user rules for a service.
readmefile with description

### CODE REVIEW

File: migrations/migrations/2_grant_priv_to_user.down.sql [hunk 1]
```diff
@@ -0,0 +1 @@
+REVOKE SELECT, INSERT, UPDATE, DELETE ON filters FROM fuser;
```
Comment (crnopster) [inline migrations/migrations/2_grant_priv_to_user.down.sql]: Is there any sense to use this?

File: migrations/migrations/1_filters_table.down.sql [hunk 2]
```diff
@@ -0,0 +1 @@
+DROP TABLE IF EXISTS filters;
```
Comment (crnopster) [inline migrations/migrations/1_filters_table.down.sql]: Is there any sense to use this?

File: migrations/README.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+DBPORT -host port for database  
+POSTGRES_USER -default owner for database  
+POSTGRES_PASSWORD -password for owner  
+POSTGRES_DB -default database name (db postgres will be also created)  
+MIGRATIONS_PATH -path to folder with .sql migrations files  
+
+## commands
+
+sudo docker run -v "$MIGRATIONS_PATH:/migrations"  --network host migrate/migrate -path=/migrations/ -database "postgresql://$POSTGRES_USER:$POSTGRES_PASSWORD@localhost:$DB_POST/$POSTGRES_DB?sslmode=disable" up
```
Comment (crnopster) [inline migrations/README.md]: that sudo..

Comment (oklymchuk) [reply migrations/README.md]: Maybe you must add --env-file ./{path to}.env to this command, becouse if you run this command not from a root where is .env docker did not find this variables

File: migrations/migrations/1_filters_table.up.sql [hunk 4]
```diff
@@ -0,0 +1,6 @@
+CREATE TABLE IF NOT EXISTS filters (
+    FTYPE VARCHAR(20) NOT NULL,
+    FCONFIGURATION JSON NOT NULL,
+    FREVERSED BOOL NOT NULL,
+    FUSER_ID VARCHAR(20) NOT NULL
```
Comment (alpaka-alpachino) [inline migrations/migrations/1_filters_table.up.sql]: Shouldn't it be unique?

File: migrations/migrations/2_grant_priv_to_user.down.sql (hunk 1 above)
Comment (maksymbilokur) [reply migrations/migrations/2_grant_priv_to_user.down.sql]: This is a part of issue "down migrations"

File: migrations/migrations/1_filters_table.down.sql (hunk 2 above)
Comment (maksymbilokur) [reply migrations/migrations/1_filters_table.down.sql]: This is a part of issue "down migrations"

File: migrations/README.md (hunk 3 above)
Comment (maksymbilokur) [reply migrations/README.md]: resolved

File: migrations/README.md [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+
+DBPORT -host port for database  
+POSTGRES_USER -default owner for database  
+POSTGRES_PASSWORD -password for owner  
+POSTGRES_DB -default database name (db postgres will be also created)  
+MIGRATIONS_PATH -path to folder with .sql migrations files  
+
+## commands
+
+docker run -v "$MIGRATIONS_PATH:/migrations"  --network host migrate/migrate -path=/migrations/ -database "postgresql://$POSTGRES_USER:$POSTGRES_PASSWORD@localhost:$DB_POST/$POSTGRES_DB?sslmode=disable" up
```
Comment (crnopster) [inline migrations/README.md]: Move this to Readme.md in the root of project

**Review by oklymchuk — DISMISSED** [review]

**Review by alpaka-alpachino — CHANGES_REQUESTED** [review]

**Review by alpaka-alpachino — DISMISSED** [review]

File: migrations/README.md (hunk 5 above)
Comment (maksymbilokur) [reply migrations/README.md]: moved

File: migrations/migrations/1_filters_table.up.sql (hunk 4 above)
Comment (maksymbilokur) [reply migrations/migrations/1_filters_table.up.sql]: no, because unique user can create more than 1 filter

**Review by crnopster — DISMISSED** [review]

**Review by alpaka-alpachino — DISMISSED** [review]

**Review by oklymchuk — DISMISSED** [review]

**Review by VasylKosiv — DISMISSED** [review]

File: storage/migrations/migrations/1_filters_table.up.sql [hunk 6]
```diff
@@ -0,0 +1,7 @@
+CREATE TABLE IF NOT EXISTS filters (
+    FNAME VARCHAR(30) UNIQUE NOT NULL,
```
Comment (dmitriyminer) [inline storage/migrations/migrations/1_filters_table.up.sql]: Why you are using 'f' prefix, and upper case?

File: README.md [hunk 7]
```diff
@@ -50,4 +50,21 @@ Filter type can be rectangular or round(in future maybe shaped)
     RectangularFilter {
         point top left, point bottom right {latitude, longitude}   // A couple of points on a map that describes margin of rectangle area.
     }
-```
+```  
+
+## MIGRATIONS
+
+Migrate with using golang/migrate default table/rules for user
+In table must be created default user with name 'fuser' without any rules with own password
```
Comment (dmitriyminer) [inline README.md]: WDYT about renaming 'fuser' to 'pf_user'?
Pattern: [app abbreviation] + '_user'

Comment (maksymbilokur) [reply README.md]: renamed

File: storage/migrations/migrations/1_filters_table.up.sql (hunk 6 above)
Comment (maksymbilokur) [reply storage/migrations/migrations/1_filters_table.up.sql]: I wanted to avoid using reserved words. now it's changed

**Review by dmitriyminer — APPROVED** [review]

**Review by crnopster — APPROVED** [review]

**Review by alpaka-alpachino — APPROVED** [review]

**Pull request merged**